### ISSUE TITLE
The menu moves the panel elements to the right outside the window

### ISSUE THREAD

**Issue by sujaataarya** [rilian-la-te/vala-panel-appmenu#48]
**Hello Rilian-la-te! Thank you very much for your program!**
  Is it possible to compile the program or supplement it with a separate patch so that in the panel where the menu is displayed, the window title is not displayed, or a word is displayed, for example "Window" or other.
  This is very necessary in such cases, when the window title is long, which results in the displacement of all panel elements beyond the visible area of ​​the display.
  I would be very grateful if there is a place for such an addition.
  _Best regards, Sergei._

  **Screenshot examples:**
  _A small window title, does not affect panel elements:_
![_2017-03-12_15-53-12](https://cloud.githubusercontent.com/assets/12985719/23831170/29820c0c-073d-11e7-8ea8-2a850fc24ff5.png)

  _The menu moves the panel elements to the right outside the display:_
![_2017-03-12_15-43-09](https://cloud.githubusercontent.com/assets/12985719/23831171/30ad9f78-073d-11e7-9d55-37e07ca03c78.png)

**Comment by rilian-la-te** [comment]
It have a workaround. Try to create desktop file for Yandex.Browser and
menu must use name of desktop file for title.

**Comment by sujaataarya** [comment]
**Спасибо за Ваш ответ!**
I created the file _/home/sergey/.local/share/applications/yb.desktop_,
but the browser does not output the name of the desktop file in the title, but translates the page name.
It would be very convenient if disable the function of displaying the name of the window in the appmenu, for example functionally.

**Comment by rilian-la-te** [comment]
Do you use Plank? How Plank shows Yandex.Browser?

**Comment by rilian-la-te** [comment]
Can you show me a desktop file for Yandex.Browser?

**Comment by sujaataarya** [comment]
Hello, thank you for your response.
I installed a plank, and it shows a simple label.

And this desktop file for Yandex.Browser?

[Desktop Entry]
Version=1.0
Name=Yandex Browser (beta)
GenericName=Web Browser
GenericName[ru]=Веб-браузер
Comment=Access the Internet
Comment[ru]=Доступ в Интернет
Exec=/usr/bin/yandex-browser-beta %U
Terminal=false
Icon=yandex-browser-beta
Type=Application
Categories=Network;WebBrowser;
MimeType=text/html;text/xml;application/xhtml_xml;image/webp;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;
Actions=NewWindow;NewPrivateWindow;

[Desktop Action NewWindow]
Name=New Window
Name[ru]=Новое окно
Exec=/usr/bin/yandex-browser-beta

[Desktop Action NewPrivateWindow]
Name=New Incognito Window
Name[ru]=Новое окно в режиме инкогнито
Exec=/usr/bin/yandex-browser-beta --incognito

**Comment by rilian-la-te** [comment]
It is strange. I need more testing. What about chromium, Firefox, libbamf and libwnck headers?

**Comment by sujaataarya** [comment]
Chromium отображает нормально.
![_2017-03-23_11-41-31](https://cloud.githubusercontent.com/assets/12985719/24235223/1acdcd7a-0fbe-11e7-895d-96bb8eb54fd5.png)

**Comment by sujaataarya** [comment]
Firefox (strange) does not show at all.
![_2017-03-23_11-42-03](https://cloud.githubusercontent.com/assets/12985719/24235298/6c8350fe-0fbe-11e7-8670-652fc2b787cc.png)

**Comment by rilian-la-te** [comment]
Please, write in English.
Ok, look like your desktop file is hard to parse for libbamf. Try to copy
chromium desktop file and replace name and executable (but borrow
executable name from ps, not from run command)

**Comment by sujaataarya** [comment]
Yandex Browser puts the entire title of the web page in the title.
![_2017-03-23_11-43-54](https://cloud.githubusercontent.com/assets/12985719/24235342/abcb3ce0-0fbe-11e7-8be7-b18b8652c6ba.png)

**Comment by sujaataarya** [comment]
I'm sorry that I did not translate immediately into English.

**Comment by sujaataarya** [comment]
Please help me to correctly understand what it means by 'ps' in the phrase (but to borrow
the name of the executable file of the SS, not the team)?

**Comment by sujaataarya** [comment]
I replaced the shortcuts of the chromium label with such

[Desktop Entry]
Name=Yandex Browser
GenericName=Web Browser
GenericName[ru]=Ð’ÐµÐ±-Ð±Ñ€Ð°ÑƒÐ·ÐµÑ€
Comment=Access the Internet
Comment[ru]=Ð”Ð¾ÑÑ‚ÑƒÐ¿ Ð² Ð˜Ð½Ñ‚ÐµÑ€Ð½ÐµÑ‚
Exec=/usr/bin/yandex-browser-beta %U
Terminal=false
Icon=yandex-browser-beta
Type=Application
Categories=GTK;Network;WebBrowser;
MimeType=text/html;text/xml;application/xhtml+xml;text/mml;x-scheme-handler/http;x-scheme-handler/https;

Renamed and copied the file yandex-browser.desktop in /usr/share/applications/
Then launched with a new shortcut.

The result did not change.

**Comment by sujaataarya** [comment]
**Hello Rilian-la-te! I really like your program.**

Is it possible to compile the additionally new program so that in the panel where the menu is displayed, the window title is not displayed. And call her "_vala-panel-appmenu-no-title_" for example. And the user will always have a choice which program suits him best! I think it will be a success!
Is it possible to implement how it is done with the menu widget on the panel in the KDE Plasma 5.9?

_Best regards, Sergey._

![screenshot_20170323_130505](https://cloud.githubusercontent.com/assets/12985719/24246126/b23801f2-0fe7-11e7-90e7-dda739a47257.png)

**Comment by rilian-la-te** [comment]
About firefox - it is normal, because firefox does not export menus on non-ubuntu.
About ps - you must determine Yandex.Browser's process name, not launcher executable. One tool for this is ps. You can use another tool.
Window will show its title (not application name from desktop file), when libbamf cannot associate process with desktop file. When it can, it will show application name from desktop file.

>  Is it possible to implement how it is done with the menu widget on the panel in the KDE Plasma 5.9?

No, because:
1. Plasma's menu widget does not take MenuModel into account at all (so, it will not work with GTK and unity-gtk-module)
2. GTK programs can have appmenu labelled with same label as program (for example, file-roller), it needed for using such programs.


And I am from Moscow, so, I understand Russian very well:)))

**Comment by rilian-la-te** [comment]
Workaround

**Comment by sujaataarya** [comment]
Hello Rilian-La-Te!
**This is one of my favorite apps.  I am infinitely grateful for the solution of my question.**
In the new version, the browser header is shortened.
Now all the panel elements remain within the screen.
**_Thank you for the excellent work and I wish you success, prosperity and good._**
_Best regards, Sergei!_

**Comment by nick87720z** [comment]
I'm currently working on support for menubar scrolling, as soon as self met this problem:
https://github.com/rilian-la-te/vala-panel-appmenu/pull/158
